docs/bioformats-release-process.rst

```diff
@@ -0,0 +1,257 @@
+-------------------
+
+Before making a new release, merge all contributions on the ``develop`` branch of `Bio-Formats`_.
+
+The first operation to perform a Maven release is to bump the version out of
+``SNAPSHOT`` using the Maven versions plugin, add and commit::
+
+    $ mvn versions:set -DnewVersion=x.y.z -DgenerateBackupPoms=false
+    $ mvn versions:set-property -Dproperty=release.version -DnewVersion=x.y.z -DgenerateBackupPoms=false
+    $ git add -u .
+    $ git commit -m "Bump release version to x.y.z"
+
+
+A signed tag must be created for the released version using :command:`git tag -s`::
+
+    $ git tag -s -m "Tag version x.y.z" vx.y.z
+
+
+Revert to ``SNAPSHOT``, add and commit::
+
+    $ mvn versions:set -DnewVersion=x.y.t-SNAPSHOT -DgenerateBackupPoms=false
+    $ mvn versions:set-property -Dproperty=release.version -DnewVersion=x.y.z-SNAPSHOT -DgenerateBackupPoms=false
+    $ git add -u .
+    $ git commit -m "Revert to snapshot"
+
+Both the ``develop`` branch as well as the tag must be pushed upstream::
+
+    $ git push origin develop vx.y.z
+
+This will trigger GitHub action builds, generate artifacts, create a GitHub release and upload
+the artifacts to the release on GitHub and to the `OME artifactory`_. 
+An hourly cron job runs on our virtual machine and copy the artifacts published on Github under `Bio-Formats Downloads`_.
+
+Close the milestone if any and add new one if needed.
+
+
+Testing the artifacts
+---------------------
+
+To test the artifacts published, 
+ - Test the `bio-formats-command-line-tools <https://docs.openmicroscopy.org/internal/testing_scenarios/GeneralRelease.html#bio-formats-command-line-tools>`_.
+ - Test `MATLAB (only) <https://docs.openmicroscopy.org/internal/testing_scenarios/GeneralRelease.html#bio-formats-matlab-octave>`_.
+ - Test `Fiji Plugin <https://docs.openmicroscopy.org/internal/testing_scenarios/GeneralRelease.html#bio-formats-imagej-fiji-testing>`_
+ - Check the schema version using the command line tools::
+      $ ./bfconvert B.ome.tiff
+      $ ./tiffcomment B.ome.tiff
+  Check that the schema version in OME-XML be ``2016-06``
+
+
+If an error occurs during the testing:
+ - Delete the release and the tag on GitHub.
+ - Delete the artifacts on `OME artifactory`_.
+ - Delete the folder corresponding to the latest release under `Bio-Formats Downloads`_, if it has already been created.
+
+Bio-Formats examples release
+----------------------------
+
+Before making a release, merge all contributions on the ``master`` branch of `Bio-Formats Examples <https://github.com/ome/bio-formats-examples>`_.
+
+The first operation to perform a Maven release is to bump the version out of
+``SNAPSHOT`` using the Maven versions plugin::
+
+    $ mvn versions:set -DnewVersion=x.y.z -DgenerateBackupPoms=false
+    $ mvn versions:set-property -Dproperty=formats-gpl.version -DnewVersion=x.y.z -DgenerateBackupPoms=false
+
+Bump the ``version`` and ``formats-gpl`` in :file:`build.gradle`.
```

Comment:
Note with https://github.com/ome/bio-formats-examples/pull/128 the proposal is to only have `formats-bsd` as a depedency

Reply:
This will have to be adjusted accordingly when the example PR is merged